File: #1598.crawler-log-folder.cpp
class Solution {
public:
    int minOperations(vector<string>& logs) {

        stack<string> st;
        for(const string& log : logs){
            if (log == "./") { 
            } else if(log == "../") {
                if (!st.empty()) {
                    st.pop();
                }
            } else {
                st.push(log);
            }
        }
        return st.size();
    }
};

File: #1598.crawler-log-folder.js
/**
 * @param {string[]} logs
 * @return {number}
 */
var minOperations = function(logs) {
    let stack = [];
    for (let i = 0; i < logs.length; i++) {
        if (logs[i] == '../'){
            stack.pop();
        }

        if (logs[i] != './' && logs[i] != '../') {
            stack.push(logs[i]);
        }
    }
    return stack.length;
};
